src/EditorFeatures/CSharpTest/MakeLocalFunctionStatic/MakeLocalFunctionStaticRefactoringTests.cs

```diff
@@ -113,7 +145,55 @@ int N()
             return x + 1;
         }
     }  
-}", parameters: new TestParameters(parseOptions: CSharp8ParseOptions));
+}",
+@"class C
+{
+    int x;
+
+    int N()
+    {
+        return AddLocal(this);
+
+        static int [||]AddLocal(C @this)
+        {
+            return @this.x + 1;
+        }
+    }  
+}", parseOptions: CSharp8ParseOptions);
+        }
+
+        [Fact, WorkItem(38734, "https://github.com/dotnet/roslyn/issues/38734")]
+        public async Task ShouldTriggerIfCapturesThisParameter3()
```

Comment:
❔ Is there already a test for cases where the local function is converted to a delegate in a manner that is not compatible with the refactoring?